pyvista/plotting/colors.py

```diff
@@ -872,6 +872,33 @@ def to_dict(self):
         """Convert to dictionary for JSON serialization."""
         return {'r': self._red, 'g': self._green, 'b': self._blue, 'a': self._opacity}
 
+    @property
+    def opacity(self):
+        """Return or set the opacity of this color in the range of ``(0-255)``.
+
+        Examples
+        --------
+        >>> import pyvista as pv
+        >>> color = pv.Color('r', opacity=0.5)
+        >>> color.opacity
+        128
+
+        Change the opacity via an attribute.
+
+        >>> color.opacity = 0.25
+        >>> color.opacity
+        64
+
+        >>> color
+        Color(name='red', hex='#ff000040', opacity=64)
+
+        """
+        return self._opacity
+
+    @opacity.setter
+    def opacity(self, value: Union[int, float, str]):
+        self._opacity = self.convert_color_channel(value)
```

Comment:
I decided against making Color instances mutable in #2175. The reason being that sometimes colors are only used as "intermediate objects" between the Pyvista user and the underlying VTK methods. See e.g.

https://github.com/pyvista/pyvista/blob/fecf347ca56221c3617a0121ec2cd603b9378439/pyvista/plotting/charts.py#L196-L216

If a user uses the color getter of a `Pen` instance and modifies the opacity, the `Color` instance will be changed, but the `SetColor` VTK method won't be called to actually update the color (as would be the case when using the color setter).
This would also break the hashing of `Color` instances (hash value is calculated from each color channel).

Reply:
ah makes sense, thanks for pointing this out. I'll modify this PR to make sure it is not mutating any `Color` objects